server/src/main/java/com/linecorp/centraldogma/server/internal/api/sysadmin/KeyManagementService.java

```diff
@@ -0,0 +1,117 @@
+        return encryptionStorageManager.wdeks();
+    }
+
+    /**
+     * Rotates the wrapped data encryption key (WDEK) for the specified project and repository.
+     */
+    @Post("/projects/{projectName}/repos/{repoName}/wdeks/rotate")
+    @Blocking
+    public CompletableFuture<Void> rotateWdek(@Param String projectName, @Param String repoName,
+                                              @Param @Default("false") boolean reencrypt) {
+        requireNonNull(projectName, "projectName");
+        requireNonNull(repoName, "repoName");
+        if (reencrypt) {
+            // TODO(minwoox): Implement re-encryption logic.
+            throw new UnsupportedOperationException();
+        } else {
+            final SecretKeyWithVersion currentDek =
+                    encryptionStorageManager.getCurrentDek(projectName, repoName);
+            return encryptionStorageManager.generateWdek().thenCompose(wdek -> {
+                // If concurrent rotation happened, only one of them will succeed to create a new WDEK because
+                // the storage manager accepts only the WDEK with the version which is exactly one greater.
+                final int newVersion = currentDek.version() + 1;
+                final WrappedDekDetails wdekDetails = new WrappedDekDetails(
+                        wdek, newVersion, encryptionStorageManager.kekId(),
+                        projectName, repoName);
+                return execute(Command.rotateWdek(projectName, repoName, wdekDetails));
+            });
+        }
+    }
+
+    /**
+     * Returns the details of the session master key.
+     * The actual key material is not included for security reasons.
+     */
+    @Get("/masterkeys/session")
+    @Blocking
+    public SessionMasterKeyDetails getSessionMasterKeyDetails() {
+        if (!encryptionStorageManager.encryptSessionCookie()) {
+            throw new IllegalStateException("Session cookie encryption is disabled.");
+        }
+        final SessionMasterKey sessionMasterKey = encryptionStorageManager.getCurrentSessionMasterKey();
+        return new SessionMasterKeyDetails(sessionMasterKey.version(),
+                                           sessionMasterKey.kekId(),
+                                           sessionMasterKey.creation());
+    }
+
+    /**
+     * Rotates the session master key.
+     */
+    @Post("/masterkeys/session/rotate")
+    @Blocking
+    public CompletableFuture<Void> rotateSessionMasterKey() {
+        if (!encryptionStorageManager.encryptSessionCookie()) {
+            throw new IllegalStateException("Session cookie encryption is disabled.");
+        }
+        final SessionMasterKey currentSessionMasterKey = encryptionStorageManager.getCurrentSessionMasterKey();
+        // If concurrent rotation happened, only one of them will succeed to create a new session master key
+        // because the storage manager accepts only the key with the version which is exactly one greater.
```

Comment:
Question) While this may be true in terms of keeping consistency, given that an exception will be thrown when checking this condition would replicas go into read-only mode?

I believe I left a similar comment previously as well: https://github.com/line/centraldogma/pull/1181/files/a10997166edba92551567a1a833936c0439ea57a#r2361488383